### ISSUE TITLE
TextDecorder issue with React Native

### ISSUE THREAD

**Issue by mariomurrent-softwaresolutions** [versatica/JsSIP#695]
https://github.com/versatica/JsSIP/blob/bd487cb5ddd3f23143f80bb3eb7e2cf23bf36eb0/lib/Transport.js#L44

This line actually breaks the JSSIP library when used with react native.

In Expo Debug Mode everything works fine, but in non debug mode TextDecoder is undefined.

**Comment by ibc** [comment]
I don't know what Expo Debug Mode is. How is it possible that TextDecoder is not defined in React Native?

**Comment by mariomurrent-softwaresolutions** [comment]
Well Expo ist just a framework used to develop react native applications.
Well I don't know why it is undefined. 

Seems like, if you wait a given a mount of time everything works fine.

**Comment by mariomurrent-softwaresolutions** [comment]
React Native v0.62.0+ is the minimum version supported where the Networking API has been made public.

The following web APIs which are not currently available in React Native:

TextEncoder
TextDecoder
ReadableStream

**Comment by ibc** [comment]
Should be fixed in jssip 3.7.3.

**Comment by mariomurrent-softwaresolutions** [comment]
Thank you very much 😊😎🤟

**Comment by nguyenduc308** [comment]
The same issue to me, when i debuging, global was not defined then it throw error invalid configuration: sockets

**Comment by mariomurrent-softwaresolutions** [comment]
> The same issue to me, when i debuging, global was not defined then it throw error invalid configuration: sockets

Same applied for me, but the error messages are misleading 😞

**Comment by ibc** [comment]
Guys, unclear whether you are reporting a new bug or not. Please clarify it and report a new issue if so.

**Comment by mariomurrent-softwaresolutions** [comment]
> Guys, unclear whether you are reporting a new bug or not. Please clarify it and report a new issue if so.

I just meant, that before you fixed the bug with the TextDecoder I received the same error message, that something is wrong with the configuration, which was actually not true ...
So also the error messages are sometimes misleading

**Comment by jeffwhelpley** [comment]
So, it looks like this bug fix created another bug because as soon as I upgraded to this version (i.e. `3.7.3`) I got an error while doing `new UA()`. It appears the line is the one that was fixed:

```
 if (global.TextDecoder) this.textDecoder = new TextDecoder('utf8');
```

The error I get is:

```
ReferenceError: global is not defined
    at new Transport (vendor.js:15191)
    at UA._loadConfig (vendor.js:18036)
    at new UA (vendor.js:17360)
```

When I revert back to `3.7.2`, there is no error and things work fine.

**Comment by ibc** [comment]
In which device?

**Comment by jeffwhelpley** [comment]
@ibc I got this error on the web (Chrome). I am not sure if it matters, but I bring in jssip via Angular (which packages the code via Webpack).

**Comment by mariomurrent-softwaresolutions** [comment]
> @ibc I got this error on the web (Chrome). I am not sure if it matters, but I bring in jssip via Angular (which packages the code via Webpack).

Maybe you need a Polyfill for it?

**Comment by ibc** [comment]
Should be fixed in JsSIP 3.7.4 released right now.

**Comment by jeffwhelpley** [comment]
OK, so as a workaround, I was able to add this to the Angular polyfills.ts file:

```
(window as any).global = window;
```

And the error went away. I don't know exactly what webpack is doing under the scenes, but just as a note it seems like when I debug the jssip code, there is an object `globalThis` available which does have TextDecoder, but not `global` (unless I throw in there the polyfill).

In any case, thanks for fixing it with 3.7.4!
